Refactor: Split out stream events

StefKors opened the pull request:
This splits out the streaming logic it separate files. It's a first pass to clean up the streaming logic for readability

Change-type: minor
Signed-off-by: Stef Kors <stef@balena.io>

grahammcculloch reviewed (changes requested):
Seems like a good change - will definitely make it easier to unit-test this functionality in the future. 

I think when you make a refactor like this it is worth adding some doc comments, e.g. above the new `streamTyping` and `streamUpdate` methods. Now that they are in their own file, the context is lost and it's not so clear just by looking at those methods what their purpose is and when they might be used and even what the arguments should be (e.g. what is `types` or `payload`).

StefKors committed:
Refactor: Split out stream events

This splits out the streaming logic it separate files. It's a first pass to clean up the streaming logic for readability

Change-type: minor
Signed-off-by: Stef Kors <stef@balena.io>
